src/main/java/com/auth0/utils/Asserts.java

```diff
@@ -4,19 +4,41 @@
 import okhttp3.HttpUrl;
 
 public abstract class Asserts {
-    public static void assertNotNull(Object value, String name) throws IllegalArgumentException {
+
+    /**
+     * Asserts that an object is not null.
+     *
+     * @param value the value to check.
+     * @param name the name of the parameter, used when creating the exception message.
+     * @throws IllegalArgumentException if the value is null or has length of zero.
+     */
+    public static void assertNotNull(Object value, String name) {
         if (value == null) {
             throw new IllegalArgumentException(String.format("'%s' cannot be null!", name));
         }
     }
 
-    public static void assertValidUrl(String value, String name) throws IllegalArgumentException {
+    /**
+     * Asserts that a value is a valid URL.
+     *
+     * @param value the value to check.
+     * @param name the name of the parameter, used when creating the exception message.
+     * @throws IllegalArgumentException if the value is null or is not a valid URL.
+     */
+    public static void assertValidUrl(String value, String name) {
         if (value == null || HttpUrl.parse(value) == null) {
             throw new IllegalArgumentException(String.format("'%s' must be a valid URL!", name));
         }
     }
 
-    public static void assertNotEmpty(Collection value, String name) throws IllegalArgumentException {
+    /**
+     * Asserts that a collection is not null and has at least one item.
```

Comment:
"and has at least one item."
 this is not being honored. Is it worth refactoring it on this PR? Or maybe just remove it from the javadocs.

Reply:
The size is being checked here in this method:
```java
if (value.size() == 0) {
     throw new IllegalArgumentException(String.format("'%s' cannot be empty!", name));
}
```

Reply:
that's true, sorry. I was only being shown the first IF check here